### ISSUE TITLE
500 errors on sources pages

### ISSUE THREAD

**Issue by gjost** [denshoproject/encyc-front#115]
> [micah] I'm getting a 500 server error when I try and look at the following link in the encyclopedia https://encyclopedia.densho.org/sources/en-denshopd-i224-00010-1/
> I'm trying to view information about the first image (Aerial view of Salinas Assembly Center, California) in the article about Salinas https://encyclopedia.densho.org/Salinas_(detention_facility)/ (edited) 
> 
> [pkikawa] Looks like the last two images are also throwing 500s
> https://encyclopedia.densho.org/sources/en-salinas-1/
> https://encyclopedia.densho.org/sources/en-salinas-2/
> [pkikawa] api looks okay: https://encyclopedia.densho.org/api/0.1/sources/en-denshopd-i224-00010-1/
> image is there: https://encyclopedia.densho.org/media/encyc-psms/en-denshopd-i224-00010-1_1.jpg

**Comment by gjost** [comment]
```
Traceback (most recent call last):
  File "/opt/encyc-front/venv/front/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/opt/encyc-front/venv/front/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/encyc-front/venv/front/lib/python3.9/site-packages/django/views/decorators/http.py", line 43, in inner
    return func(request, *args, **kwargs)
  File "/opt/encyc-front/front/wikiprox/views.py", line 135, in source
    'article_url': source.article().absolute_url(),
Exception Type: AttributeError at /sources/en-denshopd-i224-00010-1/
Exception Value: 'NoneType' object has no attribute 'absolute_url'
```
line 135         `'article_url': source.article().absolute_url(),`

Variable | Value
-- | --
encyclopedia_id | 'en-denshopd-i224-00010-1'
request | <WSGIRequest: GET '/sources/en-denshopd-i224-00010-1/'>
source | <wikiprox.models.Source "en-denshopd-i224-00010-1">
template_name | 'wikiprox/source.html'

**Comment by gjost** [comment]
For these sources the `external_url` field, which points to the article, is empty.   This needs to be fixed in the PSMS.

**Comment by gjost** [comment]
These appear to be items that we don't have DDR permissions for.  Until we have a better solution I'll just catch the error and these items just won't have links,

**Comment by gjost** [comment]
Turns out `source.headword` is the field used in `source.article()` and in these cases the field is not blank.  Also turns out that the `article_url` variables that is the goal of the crashing line is used for the back link to the Source's Article.

I ended up catching the `AttributeError` and getting the back link from `request.META['HTTP_REFERER']`.  This backlink fail if there is no referer but I can't fix everything.

Hotfixed in commit `e57b808`, merged in to `master` and installed on stage and production.
